### PR TITLE
MAP ROTATION PART 2

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Adds map rotation. This exists as a vote for players when the shuttle leaves the station (On a config option), and gives admins a verb to set the map for the next round. The default map is Cyberiad, and the "Set Next Map" verb only affects the next round, after that we go back to boxstation. 

This also introduces a benefit for people testing maps. You no longer need to recompile to test station remaps, just restart your server. This does come at the cost of an init time penalty, but its not hugely significant, and it will only go down once I kill off `/New()` further.

Yes I know this PR took so long to do, but atleast its done now, and we claimed 12345 for something good other than an insta-close meme PR 

## Why It's Good For The Game
Ability to switch maps with no external tools is a godsend

## Changelog
:cl: AffectedArc07
add: Map Rotation
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
A note on why I submitted this so early. I decieded that PR 12345 should be something truly special for the game instead of an insta-close meme PR, and the maintainers agreed.

**Comment by variableundefined** [comment]
Can't approve this yet because well, yeah.

**Comment by Ty-Omaha** [comment]
12345

**Comment by AffectedArc07** [comment]
Changes will appear once i standardise

**Comment by AffectedArc07** [comment]
Drafted now because the draft feature did not exist when this PR was made

**Comment by AffectedArc07** [comment]
The fact this managed to conflict with *one* file is astounding.

File: code/modules/admin/verbs/mapping.dm
Comment (SabreML) [inline code/modules/admin/verbs/mapping.dm:198]: This runtimes if you choose `null` for `target_map_name`.
```suggestion
	var/target_map_name = input(usr, "Select target map", "Next map", null) as null|anything in map_datums
	if(!target_map_name)
		return

	var/datum/map/TM = map_datums[target_map_name]
	SSmapping.next_map = new TM
```

**Review by SabreML — COMMENTED** [review]
Just skimmed through it a bit.

File: code/controllers/subsystem/mapping.dm
Comment (AffectedArc07) [reply code/controllers/subsystem/mapping.dm:118]: This will never happen in a prod scenario and should only appear to people on local test boxed, who will see it if they are launching with the debugger (F5), which force logs them in

File: code/__HELPERS/names.dm
Comment (AffectedArc07) [reply code/__HELPERS/names.dm:20]: For a future PR, this one is already huge

File: code/modules/admin/topic.dm
Comment (dearmochi) [inline code/modules/admin/topic.dm:2342]:
```suggestion
		var/tmsg = "<font face='Verdana' color='black'><center><img src = 'ntlogo.png'><BR><BR><BR><font size='4'><b>[SSmapping.map_datum.fluff_name]</b></font><BR><BR><BR><font size='4'>NAS Trurl Communications Department Report</font></center><BR><BR>"
```

`fluff_name` already contains NSS.

File: code/game/objects/items/devices/autopsy.dm
Comment (dearmochi) [inline code/game/objects/items/devices/autopsy.dm:77]:
```suggestion
		R.info = "<b>[SSmapping.map_datum.fluff_name] - Coroner's Report</b><br><br><b>Name of Deceased:</b> [dead_name]</br><br><b>Rank of Deceased:</b> [dead_rank]<br><br><b>Time of Death:</b> [dead_tod]<br><br><b>Cause of Death:</b> [dead_cause]<br><br><b>Trace Chemicals:</b> [dead_chems]<br><br><b>Additional Coroner's Notes:</b> [dead_notes]<br><br><b>Coroner's Signature:</b> <span class=\"paper_field\">"
```

`fluff_name` already contains NSS.

**Review by dearmochi — APPROVED** [review]
Blessed be map rotation

File: code/controllers/subsystem/mapping.dm
Comment (SabreML) [reply code/controllers/subsystem/mapping.dm:118]: I know that some people prefer running it through dream daemon which is why I mentioned, but it'll be fine in VSCode yeah.

**Review by SteelSlayer — APPROVED** [review]
LGTM

File: tools/ci/dm.sh
Comment (AffectedArc07) [inline tools/ci/dm.sh:46]: @Kyep pinging to make sure this doesn't break backend tooling, as the `-m` arg is surplus at this point, and entirely unsupported in the code now.

Comment (Kyep) [reply tools/ci/dm.sh:46]: Currently, the backend does this:
tools\ci\dm.sh paradise.dme -Mcyberiad
I assume this is no longer supported, and after this PR is merged I need to delete the "-Mcyberiad" from this line.

Comment (AffectedArc07) [reply tools/ci/dm.sh:46]: Yes that is the case, however `/datum/map/cyberiad` is the default map datum
https://github.com/ParadiseSS13/Paradise/blob/8e4f17832f6fdc51c72ebd461438b27711040cad/code/controllers/subsystem/mapping.dm#L25

Comment (Kyep) [reply tools/ci/dm.sh:46]: So what you're telling me is that I can delete the -Mcyberiad right now and it won't matter because its already the default.

Comment (AffectedArc07) [reply tools/ci/dm.sh:46]: Yep. If you dont specify a map it builds cyberiad by default
```
$ tools/ci/dm.sh paradise.dme
DM compiler version 513.1536
loading paradise.mdme
loading interface\skin.dmf
loading map_files\generic\centcomm.dmm
loading map_files\cyberiad\cyberiad.dmm
loading map_files\generic\Lavaland.dmm
saving paradise.mdme.dmb (DEBUG mode)
paradise.mdme.dmb - 0 errors, 0 warnings (1/28/21 9:07 pm)
Total time: 0:31
paradise.mdme.dmb - 0 errors, 0 warnings (1/28/21 9:07 pm)
```

**Comment by AffectedArc07** [comment]
Few minor issues discovered in the TM, aim to get them fixed soon.

**Comment by AffectedArc07** [comment]
The issue with text macros has now been fixed

**Comment by Spacemanspark** [comment]
Merge when

**Comment by DanaDririon** [comment]
When

**Review by Crazylemon64 — APPROVED** [review]

**Comment by Spacemanspark** [comment]
It finally happened

**Pull request merged**
